Lib/test/test_listcomps.py

```diff
@@ -381,6 +381,32 @@ def f():
         with self.assertRaises(UnboundLocalError):
             f()
 
+    def test_global_outside_cellvar_inside_plus_freevar(self):
+        code = """
+            a = 1
+            def f():
+                [(lambda: b) for b in [a]]
+                return b
+            x = f()
+        """
+        self._check_in_scopes(
+            code, {"x": 2}, ns={"b": 2}, scopes=["function", "module"])
```

Comment:
Let's test that the inner `b` is resolved correctly (untested):

```suggestion
                func, = [(lambda: b) for b in [a]]
                return b, func()
            x = f()
        """
        self._check_in_scopes(
            code, {"x": (2, 1)}, ns={"b": 2}, scopes=["function", "module"])
```